docs/usage/TYPE_SYNC.md

```diff
@@ -0,0 +1,120 @@
+
+- `allowUuid`
+a flag, if set to `true`, enables the user to use keys with UUID format for references. By default, it is set to `false`.
+
+Example of options usage, that sets the error and warning callbacks to output the message to the log error and warning
+streams, would look as follows:
+```java
+final Logger logger = LoggerFactory.getLogger(MySync.class);
+final TypeSyncOptions typeSyncOptions = TypeSyncOptionsBuilder.of(sphereClient)
+                                                              .errorCallBack(logger::error)
+                                                              .warningCallBack(logger::warn)
+                                                              .build();
+```
+
+#### Running the sync
+After all the aforementioned points in the previous section have been fulfilled, to run the sync:
+````java
+// instantiating a type sync
+final TypeSync typeSync = new TypeSync(typeSyncOptions);
+
+// execute the sync on your list of types
+CompletionStage<TypeSyncStatistics> syncStatisticsStage = typeSync.sync(typeDrafts);
+````
+The result of the completing the `syncStatisticsStage` in the previous code snippet contains a `TypeSyncStatistics`
+which contains all the stats of the sync process; which includes a report message, the total number of updated, created,
+failed, processed types and the processing time of the last sync batch in different time units and in a
+human readable format.
+
+````java
+final TypeSyncStatistics stats = syncStatisticsStage.toCompletebleFuture().join();
+stats.getReportMessage();
+/*"Summary: 2000 types were processed in total (1000 created, 995 updated, 5 failed to sync)."*/
+````
+
+__Note__ The statistics object contains the processing time of the last batch only. This is due to two reasons:
+ 1. The sync processing time should not take into account the time between supplying batches to the sync.
+ 2. It is not not known by the sync which batch is going to be the last one supplied.
+
+More examples of how to use the sync can be found [here](/src/integration-test/java/com/commercetools/sync/integration/types/TypeSyncIT.java).
+
+### Build all update actions
+
+A utility method provided by the library to compare a `Type` with a new `TypeDraft` and results in a list of type update actions.
+```java
+List<UpdateAction<Type>> updateActions = TypeSyncUtils.buildActions(type, typeDraft, typeSyncOptions);
+```
+
+### Build particular update action(s)
+
+Utility methods provided by the library to compare the specific fields of a `Type` and a new `TypeDraft`, and in turn builds
+ the update action. One example is the `buildChangeNameUpdateAction` which compares names:
+````java
+Optional<UpdateAction<Type>> updateAction = TypeUpdateActionUtils.buildChangeNameAction(oldType, typeDraft);
+````
+More examples of those utils for different fields can be found [here](/src/test/java/com/commercetools/sync/types/utils/TypeUpdateActionUtilsTest.java).
```

Comment:
What about `FieldDefinitionUpdateActionUtilsTest`? Should you also mention it in the docs or is it only for internal usage?

Reply:
Yes, I should mention 👍

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/304/commits/f4fc41525cbd101ff25de148431849084815c98a#diff-649a74fcd11f3ef15302f4486a9ddbd1R113